### ISSUE TITLE
Bug: consistent-type-definitions is not as humanlike as it could be

### ISSUE THREAD

**Issue by mcmire** [typescript-eslint/typescript-eslint#7847]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgMIHsA27YFdPIDeyAvgFBmiSyIqrBQKYrHmXjTxJpY4z71GzZBAAekEABMAzj2x5MAGjQMmLMsmQg4AWwgAuZNLBRQAcwDcZEkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWgBNEAM2aV8lLqgwBtSImjQO0SABoowxJAC64MAF8QJoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
interface ColorfulCircle extends Colorful, Circle {
  name: string;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-definitions": ["error", "type"],
  },
};
```

### Expected Result

```
type ColorfulCircle = Colorful & Circle & {
  name: string
};
```

### Actual Result

```
type ColorfulCircle = {
  name: string;
} & Colorful & Circle;
```

### Additional Info

I realize this is not a bug per se, more of an improvement that produces a result that is more likely to make sense for people who generally write type aliases by hand. That said, if anyone else feels like they would prefer the "expected result" above, I have a PR ready to go.

**Comment by JoshuaKGoldberg** [comment]
Ah, this is really a style preference separate from the consistent-type-definitions rule. Sorting of type constituents is handled by https://typescript-eslint.io/rules/sort-type-constituents. Rule fixers generally only fix for the rule's specific issue, not anything configured in other rules.

**Comment by mcmire** [comment]
@JoshuaKGoldberg Hmm, interesting. The `sort-type-constituents` rule seems to sort constituents alphabetically. That isn't quite what I want here — I would just like the "child type" to go last when the fixer applies the conversion, fitting the same sort of mental model as interfaces, where a "child" interface is applied "on top of" its "parents" (I realize I am mixing metaphors here, and that's not really how interfaces work anyway, but hopefully you see what I mean). I am not sure that the `sort-type-constituents` rule would know to do that once the conversion has already taken place.

**Comment by JoshuaKGoldberg** [comment]
That's a very specific preference. Not saying it's right or wrong, just - not in scope for something we'd want to spend time on in this project. We've found that style preferences can be _very_ difficult to get right, in large part because "right" means "configurable to every popular permutation" (let alone the less common ones). In fact, ESLint in general is moving away from stylistic rules: https://eslint.org/blog/2020/05/changes-to-rules-policies/#what-are-the-changes. I personally am planning on starting discussions soon about find another repository for `sort-type-constituents`, such as [`eslint-plugin-perfectionist`](https://github.com/azat-io/eslint-plugin-perfectionist).

I really like how you're thinking about it though 😄. Good metaphor(s). It'd be fantastic if you could either contribute this to an existing community plugin (maybe [`eslint-plugin-perfectionist`](https://github.com/azat-io/eslint-plugin-perfectionist)? not sure...) and/or write this as a standalone ESLint plugin. That'd let you iterate on it outside of our limited release cycle & approval process. Then, if it ends up being a very popular standard and in scope of this project, we can take it in. Either way the docs at https://typescript-eslint.io/developers/custom-rules are likely to be very helpful.
